src/main/java/com/ericsson/ei/jmespath/IncompletePathFilterFunction.java

```diff
@@ -0,0 +1,159 @@
+package com.ericsson.ei.jmespath;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.github.wnameless.json.flattener.JsonFlattener;
+
+import io.burt.jmespath.Adapter;
+import io.burt.jmespath.JmesPathType;
+import io.burt.jmespath.function.ArgumentConstraints;
+import io.burt.jmespath.function.BaseFunction;
+import io.burt.jmespath.function.FunctionArgument;
+
+public class IncompletePathFilterFunction extends BaseFunction {
+
+    public IncompletePathFilterFunction() {
+        super(ArgumentConstraints.listOf(ArgumentConstraints.typeOf(JmesPathType.OBJECT), ArgumentConstraints.typeOf(JmesPathType.STRING)));
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see io.burt.jmespath.function.BaseFunction#callFunction(io.burt.jmespath.Adapter, java.util.List)
+     * 
+     * This takes JSON object and a key. The key can contain the whole path, parts of path, or only a simple key.
+     * It search through the whole object after the values that have same key and returns a Map that contains
+     * a key and a list of all found values.
+     */
+    @Override
+    protected <T> T callFunction(Adapter<T> runtime, List<FunctionArgument<T>> arguments) {
+        T value1 = arguments.get(0).value();
+        T value2 = arguments.get(1).value();
+
+        String object = runtime.toString(value1);
+        String key = runtime.toString(value2);
+
+        T result = null;
+        ArrayList<String> arrayResult = filterObjectWithIncompletePath(object, key);
+        if (arrayResult == null || arrayResult.isEmpty()) {
+            result = runtime.createString(null);
+        } else if (arrayResult.size() == 1) {
+            result = runtime.createString(arrayResult.get(0));
+        } else {
+            result = runtime.createString(arrayResult.toString());
+        }
+
+        return result;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * Flatten an object and creates a list with the parts of search key. Returns array that contains filtered values.
+     */
+    private ArrayList<String> filterObjectWithIncompletePath(String object, String key) {
+        Map<String, Object> flattJson = JsonFlattener.flattenAsMap(object);
+        ArrayList<String> resultArray = new ArrayList<String>();
+        List<String> keyParts = Arrays.asList(key.split("\\."));
+        looptThroughAllPaths(resultArray, flattJson, keyParts);
+
+        return resultArray;
+    }
+
+    private void looptThroughAllPaths(ArrayList<String> resultArray, Map<String, Object> flattJson, List<String> keyParts) {
```

Comment:
Minor misspelling. An extra t has found its way into that function name.

Reply:
Done